packages/amplify-util-mock/src/api/lambda-trigger-handler.ts

```diff
@@ -0,0 +1,134 @@
+
+    await pollForRecords(context, streamArn, streams, lambdaTriggerName);
+}
+
+/**
+ * Shards are ephemeral and last for ~15 mins. This method fetches the latest Shard iterator.
+ * @param streamArn DDB stream ARN
+ * @param streams DDB streams client
+ * @returns latest active shard iterator
+ */
+export const getLatestShardIterator = async (streamArn: string, streams: DynamoDBStreams): Promise<string> => {
+    const stream = await streams
+        .describeStream({ StreamArn: streamArn })
+        .promise();
+
+    if (!stream) {
+        throw new Error(`Local DynamoDB stream with ARN ${streamArn} cannot be found`);
+    }
+
+    const { ShardId: shardId } =
+        stream.StreamDescription.Shards.filter(
+        x =>
+            x.SequenceNumberRange &&
+            x.SequenceNumberRange.StartingSequenceNumber &&
+            !x.SequenceNumberRange.EndingSequenceNumber
+        )[0] || {}
+
+    if (!shardId) {
+        throw new Error('There is no shard that is open');
+    }
+
+    const { ShardIterator: start } = await streams
+        .getShardIterator({
+            StreamArn: streamArn,
+            ShardId: shardId,
+            ShardIteratorType: 'LATEST'
+        })
+        .promise();
+    
+    return start;
+}
+
+/**
+ * Gets the latest available records from a stream. Refreshes the shard iterator if the shard has expired.
+ * @param shardIterator current shard iterator
+ * @param streamArn DDB stream ARN
+ * @param streams DDB streams client
+ * @returns latest available records from stream
+ */
+export const getStreamRecords = async (shardIterator: string, streamArn: string, streams: DynamoDBStreams) => {
+    const shardIteratorCopy = shardIterator;
+    try {
+        const data = await streams.getRecords({ ShardIterator: shardIteratorCopy }).promise();
+        return { data: data, shardIterator: shardIteratorCopy };
+    } catch (error) {
+        console.log('Re-Trying with a new shard');
+        const latestShardIterator = await getLatestShardIterator(streamArn, streams);
+        const data = await streams.getRecords({ ShardIterator: latestShardIterator }).promise();
+        return { data: data, shardIterator: latestShardIterator };
+    }
+}
+
+/**
+ * Checks if there are any new DDB records available 
+ * via stream to be processed in the trigger
+ */
+export const pollForRecords = async(context: $TSContext, streamArn: string, streams: DynamoDBStreams, lambdaTriggerName: string) => {
+    let shardIterator = await getLatestShardIterator(streamArn, streams);
+    while (shardIterator) {
+        await getStreamRecords(shardIterator, streamArn, streams).then( async (result) => {
+            const data = result.data;
+            shardIterator = result.shardIterator;
+
+            if (data.Records.length) {
+                // when the records are available to be processed, trigger the local lambda
+                await invokeLambda(context, lambdaTriggerName, data).then( () => {
```

Comment:
we wouldn't want to handle exceptions here since there is no retry. Since this is a mock setup, we would want to quit the async task if there is any error invoking the lambda so that the developer can fix it and re-run.